webapp/app/js/compare.js

```diff
@@ -0,0 +1,403 @@
+                                                     response.data[signature],
+                                                     optionCollectionMap,
+                                                     $stateParams.pgo,
+                                                     $stateParams.e10s);
+
+                if (seriesSummary != null && seriesSummary.signature !== undefined) {
+                  newSeriesList.push(seriesSummary);
+
+                  if ($scope.platformList.indexOf(seriesSummary.platform) === -1) {
+                    $scope.platformList.push(seriesSummary.platform);
+                  }
+
+                  if ($scope.testList.indexOf(seriesSummary.name) === -1) {
+                    $scope.testList.push(seriesSummary.name);
+                  }
+                }
+              });
+              $scope.testList.sort();
+              $scope.platformList.sort();
+
+
+              $q.all(newSeriesList.map(function(series) {
+                return $http.get(signatureURL + "&signatures=" + series.signature).then(function(response) {
+                  response.data.forEach(function(data) {
+                    newRawResultsMap[data.series_signature] = calculateStats(data.blob, $scope.newResultSetID);
+                    newRawResultsMap[data.series_signature].name = series.name;
+                    newRawResultsMap[data.series_signature].platform = series.platform;
+                  });
+                });
+              })).then(function () {
+                $scope.dataLoading = false;
+                displayResults(rawResultsMap, newRawResultsMap);
+              });
+            });
+          });
+        }
+      );
+    }
+
+    //TODO: put this into a generic library
+    function calculateStats(perfData, resultSetID) {
+      var geomeans = [];
+      var total = 0;
+      perfData.forEach(function(pdata) {
+        if (pdata.result_set_id === resultSetID) {
+          geomeans.push(pdata.geomean);
+          total += pdata.geomean;
+        }
+      });
+
+      var avg = total / geomeans.length;
+      var sigma = stddev(geomeans, avg);
+      return {'geomean': avg.toFixed(2), 'stddev': sigma.toFixed(2), 'runs': geomeans.length};
+    }
+
+    //TODO: put this into a generic library
+    function isReverseTest(testName) {
+      var reverseTests = ['dromaeo_dom', 'dromaeo_css', 'v8_7', 'canvasmark'];
+      var found = false;
+      reverseTests.forEach(function(rt) {
+        if (testName.indexOf(rt) >= 0) {
+          found = true;
+        }
+      });
+      return found;
+    }
+
+    function displayResults(rawResultsMap, newRawResultsMap) {
+      var counter = 0;
+      var compareResultsMap = {};
+
+      $scope.testList.forEach(function(testName) {
+        if (counter > 0 && compareResultsMap[(counter-1)].headerColumns==2) {
+          counter--;
+        }
+
+        //TODO: figure out a cleaner method for making the names a header row
```

Comment:
I think it would honestly be better to just hardcode most of the header inside the .html. For the stuff that changes, let's store a bunch of seperate properties and modify the header inside the html accordingly. Then just do the ng-repeat on the individual rows.

I guess this would mean also breaking out each of the subgroups, but that shouldn't be so bad, I don't think? I'm imagining an object like this:

$scope.compareGroups = [ { header: { ... header properties ... }, { data: { ... data properties ... }, ... ]